libminifi/src/core/ProcessSession.cpp

```diff
@@ -290,6 +290,33 @@ void ProcessSession::append(const std::shared_ptr<core::FlowFile> &flow, OutputS
   }
 }
 
+namespace {
+// FlowFileStream uses the flowfile's size instead of the stream size (so it will not be over-read)
+class FlowFileStream : public io::BaseStream {
```

Comment:
the only thing we use from the flowfile is the offset and size, I don't think we need to store the flowfile, we could generalize this `FlowFileStream` to a `StreamSlice` taking an offset and size

Reply:
Good idea.
I also changed seek() to work properly from the beginning of the slice (instead of the stream).

Added also a couple more unit tests. https://github.com/apache/nifi-minifi-cpp/pull/1177/commits/0a442efb33954be698d0ed5daa5148ab5f3bd02b